lib/ansible/modules/cloud/huawei/hwc_network_vpc.py

```diff
@@ -164,9 +164,10 @@ def main():
             expect = _get_editable_properties(module)
             current_state = response_to_hash(module, fetch)
             if is_different(expect, current_state):
-                fetch = update(session, self_link(session), [200])
-                fetch = response_to_hash(module, fetch.get('vpc'))
-                changed = True
+            	if not module.check_mode:
+               	    fetch = update(session, self_link(session), [200])
+                    fetch = response_to_hash(module, fetch.get('vpc'))
+                    changed = True
```

Comment:
The `changed = True` must be outside the `if`, otherwise checkmode doesn't work.

Reply:
Also, for check more to work properly, you have to put
```
            fetch = create(session, collection(session), [200])
            fetch = response_to_hash(module, fetch.get('vpc'))
```
below into an `if:`.

Reply:
done

Reply:
In case you changed this, you still need to push the change to GitHub. (This also has to be done for the `delete()` call.)

Reply:
Done